test/tests.js

```diff
@@ -123,253 +131,353 @@ describe('Optimizely Forwarder', function () {
+});
 
-        done();
+    // Optimizely Full Stack Tests
+    describe('Full Stack', function () {
+        beforeEach(function() {
+            // Mock Optimizely Full Stack Datafile
+            window.optimizelyDatafile =
+            {
+                "version": "4",
+                "projectId": "18390990327",
+                "variables": [],
+                "featureFlags": [
+                  {
+                    "experimentIds": [],
+                    "rolloutId": "18375930463",
+                    "variables": [
+                      {
+                        "defaultValue": "",
+                        "type": "string",
+                        "id": "18375860245",
+                        "key": "f2variable1"
+                      }
+                    ],
+                    "id": "18360310384",
+                    "key": "featurename2"
+                  },
+                  {
+                    "experimentIds": [],
+                    "rolloutId": "18389000125",
+                    "variables": [
+                      {
+                        "defaultValue": "",
+                        "type": "string",
+                        "id": "18358530123",
+                        "key": "f1variable1"
+                      }
+                    ],
+                    "id": "18381870404",
+                    "key": "featurename1"
+                  }
+                ],
+                "experiments": [],
+                "accountId": "7015510640",
+                "events": [
+                  {
+                    "experimentIds": [],
+                    "id": "18356890341",
+                    "key": "eventKey2"
+                  },
+                  {
+                    "experimentIds": [],
+                    "id": "18391220198",
+                    "key": "eventKey1"
+                  }
+                ]
+              }
+            // Include any specific settings that are required for initializing your SDK here
+            var sdkSettings = {
+                projectId: 'LYLgZJqZzFKd5SaNLcQRc'
+            };
+            // You may require userAttributes or userIdentities to be passed into initialization
+            var userAttributes = {
+                color: 'green'
+            };
+            var userIdentities = [{
+                Identity: 'customerId',
+                Type: IdentityType.CustomerId
+            }, {
+                Identity: 'email',
+                Type: IdentityType.Email
+            }, {
+                Identity: 'facebook',
+                Type: IdentityType.Facebook
+            }];
+
+            mParticle.forwarder.init(sdkSettings, reportService.cb, true, null, userAttributes, userIdentities);
+        });
+
+        it('should log a custom event', function(done) {
+            mParticle.forwarder.process({
+                EventDataType: MessageType.PageEvent,
+                EventName: 'eventKey1',
+                UserId: 'tkatz@mparticle.com',
```

Comment:
This is probably a relic from when you were setting up before, but `UserId` is not a proper key here.  i'd remove.

Reply:
Ah, yes. The one test in place is messy and doesn't map correctly based on code updates. I'll get this restructured and actual tests written today.